src/backend/cpu/image.cpp

```diff
@@ -40,8 +42,44 @@ void copy_image(const Array<T> &in, const fg::Image* image)
     CheckGL("In CopyArrayToPBO");
 }
 
+template<typename T>
+Array<float> moments(const Array<T> &in, const af_moment moment)
```

Comment:
Put this in a new file called moments.cpp. Image.cpp is for the image() function. Don't mix them up. Also, if graphics is disabled, this will cause error.